differ/mutationDiffer.go

```diff
@@ -930,10 +930,17 @@ func (b *batch) send() error {
 	for _, fetchItem := range b.fetchList {
 		b.fetchItemAndStoreResult(fetchItem)
 	}
-
+	getBody := false
+	if b.dw.differ.compareType == base.MutationCompareTypeBodyOnly ||
+		b.dw.differ.compareType == base.MutationCompareTypeBodyAndMeta {
+		getBody = true
+	}
 	doneChan := make(chan bool, 1)
-	go utils.WaitForWaitGroup(&b.waitGroup, doneChan)
-
+	if getBody {
+		go utils.WaitForWaitGroup(&b.waitGroupBodyCompare, doneChan)
+	} else {
+		go utils.WaitForWaitGroup(&b.waitGroupMetaCompare, doneChan)
```

Comment:
Why do you separate one waitgroup into two distinct ones?
Doesn't that make it easier to introduce a race by missing Done or missing Add or missing Wait? Right now I only see that it's either body or meta, so one is sitting idle /not used while the other one is used, correct?

Reply:
Please ignore the above change.
I was trying out something and I forgot to revert back before pushing this change.
One wait group is sufficient and I'll checkin again.